### ISSUE TITLE
p-dialog is positioned wrong on first time

### ISSUE THREAD

**Issue by DarioN1** [primefaces/primeng#5584]
Hi Guys,

I don't know why, p-dialog component is opening wrong during the first time.

If then I close it and I reopein it, it is positioned properly...

How can I fix it ?

May be it depents on the html template I'm using ( adminlte2 ), do I have some css property to override or to set "!important" ?

Thanks to support

**Comment by dicbrus** [comment]
Please provide some fiddle/plunkr.
You issue can be related to the content of the dialog, if you show it dynamically.
Might be the case that when window is positioned, its content is empty.

**Comment by jsilveira2** [comment]
I had this same problem, my modal was opening on the right side of the screen the first time I opened it. Researching a little I found two probable reasons for this to happen: 

- First: I had not defined the width of the dialog and this was causing the problem in my case. From the moment I set the width the problem has stopped occurring.

- Second: Your dialog may be inside a NgIf as in the case of that person in StackOverflow: https://stackoverflow.com/questions/40800875/primeng-p-dialog-not-being-centered-on-first-display

-  Third: The screen you are trying to load inside the modal has many things to load and the modal ends up opening before the screen even loads its components. In this case try using a setTimeout.

Hope this helps. 
Ps: Sorry for my bad english.
